src/terminal/adapter/DispatchTypes.hpp

```diff
@@ -109,31 +111,33 @@ namespace Microsoft::Console::VirtualTerminal
             return _value >= 0;
         }
 
-        constexpr size_t value() const noexcept
+        constexpr VTInt value() const noexcept
         {
             return _value;
         }
 
-        constexpr size_t value_or(size_t defaultValue) const noexcept
+        constexpr VTInt value_or(VTInt defaultValue) const noexcept
         {
-            return has_value() ? _value : defaultValue;
+            // A negative value indicates that the parameter was omitted.
+            return _value < 0 ? defaultValue : _value;

```

Comment:
Turns out that MSVC is really picky about how you write conditions. ô_O
Good: `_value < 0 ? defaultValue : _value`
Bad: `_value >= 0 ? _value : defaultValue`

In gcc/clang it produces the same assembly. Well, honestly, this change is so insanely trivial it doesn't really matter, but I was curious and here we are. I personally prefer this code as I wrote elsewhere already as it's immediately obvious what this line does (arithmetically) without having to check other functions. But... eh.